pyzim/pyzim.pyx

```diff
@@ -410,10 +214,26 @@ cdef class ZimCreator:
         new_metadata = {pascalize(key): value for key, value in kwargs.items()}
         self._metadata.update(new_metadata)
 
+    def _update_article_counter(self, ZimArticle article):
+        # default dict update
+        self._article_counter[article.mimetype] += 1
+
+    def add_article(self, ZimArticle article not None):
+        if not article.can_write:
+            raise RuntimeError("Article is not good for writing")
+
+        # Make a shared pointer to ZimArticleWrapper from the ZimArticle object (dereference internal c_article)
+        cdef shared_ptr[ZimArticleWrapper] art = make_shared[ZimArticleWrapper](dereference(article.c_article));
+        try:
+            self.c_creator.addArticle(art)
+        else:
+            if not article.is_redirect:
+                self._update_article_counter(article)
```

Comment:
A `try` without an `except` is mostly useless.
It is even not valid python.

Reply:
Mistake.
```python3
try:
            self.c_creator.addArticle(art)
        except:
            raise
        else:
            if not article.is_redirect:
                self._update_article_counter(article)
```